### PR TITLE
docs: add structure diagram

### PR DESCRIPTION
**Pull request by tlambert03**
this adds a diagram that looks like this to the docs

Screenshot 2022-12-22 at 10 47 10 AM

### CODE REVIEW

File: docs/USAGE.md [hunk 1]
```diff
@@ -132,6 +132,41 @@ canvas.width = 1000 # calls canvas._backend._vis_set_width(1000)
...
+    VisModel --* BackendAdaptorProtocol : controls a
+    BackendAdaptorProtocol <-- Backend1Adaptor : implements
+    BackendAdaptorProtocol <-- Backend2Adaptor : implements
+    <<Interface>> BackendAdaptorProtocol
+
+    VisModel <-- Canvas : inherits
+    Canvas --* CanvasAdaptorProtocol
+    BackendAdaptorProtocol <-- CanvasAdaptorProtocol : inherits
+    CanvasAdaptorProtocol <-- VispyCameraAdaptor : implements
+    CanvasAdaptorProtocol <-- PygfxCameraAdaptor : implements
```
Comment (alisterburt) [inline docs/USAGE.md:149]:
```suggestion
    CanvasAdaptorProtocol <-- VispyCanvasAdaptor : implements
    CanvasAdaptorProtocol <-- PygfxCanvasAdaptor : implements
```

**Review by alisterburt — COMMENTED** [review]
pretty! thing there was a small error but otherwise great stuff!!

File: docs/USAGE.md (hunk 1 above)
Comment (tlambert03) [reply docs/USAGE.md]: good catch!  (yeah, I changed the example from camera to canvas :)

**Comment by tlambert03** [comment]
i think the poor style (clipping words) is a github CSS problem.  not gonna worry about it now

**Pull request merged**